gui/gm-editor.lua

```diff
@@ -135,16 +140,56 @@ function GmEditorUi:init(args)
     local mainList=widgets.List{view_id="list_main",choices={},frame = {l=1,t=3,yalign=0},on_submit=self:callback("editSelected"),
         on_submit2=self:callback("editSelectedRaw"),
         text_pen=COLOR_GREY, cursor_pen=COLOR_YELLOW}
+	self.autoupdateLabel = widgets.Label{text={{gap=1,text="Auto-Update stopped...",key=keybindings.autoupdate.key,key_sep = '()'}}, view_id = 'lbl_autoupdate',frame = {l=0,t=0,yalign=0}}
     local mainPage=widgets.Panel{
         subviews={
             mainList,
             widgets.Label{text={{text="<no item>",id="name"},{gap=1,text="Help",key=keybindings.help.key,key_sep = '()'}}, view_id = 'lbl_current_item',frame = {l=1,t=1,yalign=0}},
-            widgets.EditField{frame={l=1,t=2,h=1},label_text="Search",key=keybindings.start_filter.key,key_sep='(): ',on_change=self:callback('text_input'),view_id="filter_input"}}
+            widgets.EditField{frame={l=1,t=2,h=1},label_text="Search",key=keybindings.start_filter.key,key_sep='(): ',on_change=self:callback('text_input'),view_id="filter_input"},
+			self.autoupdateLabel}
         ,view_id='page_main'}
 
     self:addviews{widgets.Pages{subviews={mainPage,helpPage},view_id="pages"}}
     self:pushTarget(args.target)
 end
+function GmEditorUi:verifyStack(args)
+	local failure = false
+
+	local last_good_level = nil
+
+	for i, level in pairs(self.stack) do
+		local obj=level.target
+
+		local keys = level.keys
+		local selection = level.selected
+		local sel_key = keys[selection]
+		local next_by_ref
+		local status, _ = pcall(
+			function()
+				next_by_ref = obj[sel_key]
```

Comment:
in addition to checking whether the key exists, can we also check to make sure it points to the same place it used to? for example, will this detect when the element at n-1 was deleted and everything just moved down a slot and the struct that we're looking at is no longer at index n?

Reply:
This was meant to be addressed by next_in_stack - see below